virtcontainers/kata_agent.go

```diff
@@ -1230,8 +1239,11 @@ func (k *kataAgent) createContainer(sandbox *Sandbox, c *Container) (p *Process,
 	}
 
 	// Ask to the shim to print the agent logs, if it's the process who monitors the sandbox and use_vsock is true (no proxy)
+	// Don't read the console socket if agent debug console is enabled.
 	var consoleURL string
-	if sandbox.config.HypervisorConfig.UseVSock && c.GetAnnotations()[vcAnnotations.ContainerTypeKey] == string(PodSandbox) {
+	if sandbox.config.HypervisorConfig.UseVSock &&
+		c.GetAnnotations()[vcAnnotations.ContainerTypeKey] == string(PodSandbox) &&
+		!k.hasAgentDebugConsole(sandbox) {
```

Comment:
Another random thought: might be good to add a log call is `k.hasAgentDebugConsole()` returns `true` so it's clear someone is already connected?

Reply:
Yes, indeed. While at it, I found that the proxy also might be reading the debug console. I've updated the PR to fix it as well. PTAL.